front/learn-next-js/src/app/components/ScheduleTable.tsx

```diff
@@ -0,0 +1,46 @@
+import { promises as fs } from 'fs';
+
+
+async function CreateTable(schedule: []) {
+    const scheduleEnries = Object.entries(schedule)
+    const firstDayInSchedule = scheduleEnries[0]
+    const scheduleDates = scheduleEnries.map((day) => day[0])
+  
+    const tHeaders = scheduleDates.map((header) => <th className="p-1 text-black">{header}</th>)
+    // the firstDayInDictionary is a list containing a stringify date and a dictionary, the dictionary containe a "shift type" as a key (string) and a string representing the employee and the shift's period of time
+    const dayInfoIndex = 1
+    const shiftsTypes = Object.keys(firstDayInSchedule[dayInfoIndex])
+  
+    const rowHeaders = shiftsTypes.map((header) => <th className="p-1 text-black">{header}</th>)
+  
+    const tBody = new Array
+  
+    for(var i = 0; i < shiftsTypes.length; i++) {
+      tBody.push(
+        <tr className="divide-x-2 divide-slate-600">
+          {rowHeaders[i]}
+          {/* day[1][shiftTypes[i]] represents the information on thet "cell" in the schedule*/}
```

Comment:
the way you dont have strongly typed APIs really screws you over here. next PR must be adding OpenAPI and npm package generation